lib/babe/verify.go

```diff
@@ -109,6 +112,54 @@ func (b *Session) verifyAuthorshipRight(slot uint64, header *types.Header) (bool
 		return false, fmt.Errorf("could not verify signature")
 	}
 
-	// TODO: check if the producer has equivocated, ie. have they produced a conflicting block?
+	// #553 check if the producer has equivocated, ie. have they produced a conflicting block?
+	hashes, err := b.blockState.GetAllHashesForParentDepth(header)
+	if err != nil {
+		return false, err
+	}
+
+	log.Debug("Got AllHashesForParentDepth", "len(hashes)", len(hashes))
+
+	for key, v := range hashes {
+		//get header
+		currentHeader, err := b.blockState.GetHeader(key)
+		if err != nil {
+			return false, fmt.Errorf("could not verify digests, key %s, depth, %d", key, v)
```

Comment:
maybe don't error but just `continue` instead.

Reply:
would be nice to log error as debug if you continue as noot suggests

Reply:
ok